deploy/firesim

```diff
@@ -26,6 +27,25 @@ from util.streamlogger import StreamLogger
 ## below are tasks that users can call
 ## to add a task, add it here and to the choices array
 
+"""Registry of tasks that use `BuildConfigFile`"""
+BUILD_TASKS: Dict[str, Callable] = {}
```

Comment:
Why is this preferred to using `globals()`?

Reply:
Actually, I see why this is needed, we need this to call the function in the `pytests`. Is there a reason to split this up into `BUILD_` and `RUN_` or is that mainly for organization?

Reply:
They are split because if `args.task` is in `BUILD_TASKS` then you make a `BuildConfigFile(args)` and pass that into calling the task.  Likewise for `RUN_TASKS` you make a `RuntimeConfig(args)` and pass that into the task.